app/controllers/organisations_controller.rb

```diff
@@ -103,7 +103,8 @@ def logs
         end
 
         format.csv do
-          send_data unpaginated_filtered_logs.to_csv, filename: "logs-#{@organisation.name}-#{Time.zone.now}.csv"
+          EmailCsvJob.perform_later(current_user, search_term, filters, "logs-#{Time.zone.now}.csv")
```

Comment:
This needs to be moved out to a POST + confirm flow for the organisation logs the same way it is for the main logs page. I could use some direction on how best to do that without just duplicating the ERB + routes (Rails helpers etc. in `_log_list.html.erb`?)

Reply:
I've got an implementation of this. Not 100% happy, but welcome either suggestions on how to improve, or agreement that it's a reasonable approach.